docs/decisions/0024-multi-modal-requirements.md

```diff
@@ -0,0 +1,296 @@
+        string prompt,
+        dynamic? requestSettings = null,
+        CancellationToken cancellationToken = default);
+
+    IAsyncEnumerable<IChatStreamingResult> GetStreamingChatCompletionsAsync(
+        string prompt,
+        dynamic? requestSettings = null,
+        CancellationToken cancellationToken = default);
+}
+```
+
+#### Scoping for V1.0.0
+
+To keep V1.0.0 scoped, we just need to ensure the v1.0.0 interfaces have the methods that require a single string as input so they
+can be called by prompt functions. The interfaces that must be updated are:
+
+- ITextCompletion
+- IChatCompletion
+- And _maybe_ IAIService
+
+_All other services (e.g., IImageGeneration) are marked experimental, so they can be ignored for right now._
+
+### Requirement #3: Make it easy to select a service with a required return type
+
+Today, each of our service interfaces have a unique return type:
+
+- `ITextCompletion`: `ITextResult`
+- `IChatCompletion`: `IChatResult`
+- `IImageGeneration`: `IImageResult`
+
+This makes it easy for the `IAIServiceSelector` to select the correct service because there's a one-to-one mapping.
+
+#### The problem
+
+In the future, however, there will likely be multiple service interfaces that return the same type. For example, in HuggingFace there are
+at least 15 different service types that can return text back (i.e., `ITextResult`). Most (if not all) of these service types will require
+different input parameters, so they cannot reuse the `ITextCompletion` interface. Therefore, we'll likely need to introduce the following
+interfaces that have the same return type:
+
+- `IImageToTextTask`: `ITextResult`
+- `IVisualQuestionAnsweringTask`: `ITextResult` and `IAnswerResult`
+- `ITextClassificationTask`: `ITextResult` and `IClassificationResult`
+- `ITokenClassificationTask`: `ITextResult` and `IClassificationResult`
+- `ITableQuestionAnsweringTask`: `ITextResult` and `ITableAnswerResult`
+- `IQuestionAnsweringTask`: `ITextResult`
+- `IZeroShotClassificationTask`: `ITextResult` and `IClassificationResult`
+- `ITranslationTask`: `ITextResult`
+- `ISummarizationTask`: `ITextResult`
+- `IConversationTask`: `ITextResult` and `IChatResult`
+- `ITextGenerationTask`: `ITextResult`
+- `IText2TextGenerationTask`: `ITextResult`
+- `IFillMaskTask`: `ITextResult` and `IMaskResult`
+- `ISentenceSimilarityTask`: `ITextResult` and `ISimilarityResult`
+
+It is unrealistic to expect the `IAIServiceSelector` to enumerate all of these interfaces _just_ to find a service that returns text.
+
+#### Ideal developer experience
+
+Ideally, the `IAIServiceSelector` should be able to easily deduce the supported output types of a service in a single line of code.
+
+```csharp
+private sealed class CustomAIServiceSelector : IAIServiceSelector
+{
+    public (T?, PromptExecutionSettings?) SelectAIService<T>(Kernel kernel, ContextVariables variables, KernelFunction function) where T : class, IAIService
```

Comment:
We still have the problem of how to determine the correct `T` to use.

If we want we can add a new abstraction e.g. `IAIMultiModelServiceSelector` in the future which determines the type and then the service and request settings. It could wrap the existing service selector to be compatible.